shared/chat/conversations-list/container.js

```diff
@@ -1,138 +1,37 @@
-      const unreadCount = 0
-      const participants = participantFilter(users, this.props.you)
-      const isSelected = this.props.selectedConversation === conversationIDKey
-      const isMuted = false
-      const rekeyInfo = null
-      const timestamp = formatTimeForConversationList(Date.now(), this.props.nowOverride)
-      const snippet = ''
-      const onSelectConversation = this.props.onSelectConversation
-
-      return {
-        conversationIDKey,
-        isMuted,
-        isSelected,
-        onSelectConversation,
-        participants,
-        rekeyInfo,
-        snippet,
-        timestamp,
-        unreadCount,
-      }
-    }).toList()
+const createImmutableEqualSelector = createSelectorCreator(defaultMemoize, I.is)
 
-    const realRows = this.props.inbox.map(conversation => {
-      const conversationIDKey: ConversationIDKey = conversation.get('conversationIDKey')
-      const unreadCount = this.props.conversationUnreadCounts.get(conversationIDKey)
-      const participants = participantFilter(conversation.get('participants'), this.props.you)
-      const isSelected = this.props.selectedConversation === conversationIDKey
-      const isMuted = conversation.get('muted')
-      const rekeyInfo = this.props.selectedConversation && this.props.rekeyInfos.get(conversationIDKey)
-      const timestamp = formatTimeForConversationList(conversation.get('time'), this.props.nowOverride)
-      const snippet = conversation.get('snippet')
-      const onSelectConversation = this.props.onSelectConversation
-
-      return {
-        conversationIDKey,
-        isMuted,
-        isSelected,
-        onSelectConversation,
-        participants,
-        rekeyInfo,
-        snippet,
-        timestamp,
-        unreadCount,
-      }
-    })
-
-    const rows = pendingRows.concat(realRows).map(props => ({
-      ...props,
-      ...this._derivedProps(props.rekeyInfo, props.unreadCount, props.isSelected),
-    }))
-
-    return <ConversationList
-      children={this.props.children}
-      onNewChat={this.props.onNewChat}
-      onSelectConversation={this.props.onSelectConversation}
-      rows={rows}
-    />
+const filteredInbox = createImmutableEqualSelector(
+  [getInbox, getSupersededByState, getAlwaysShow],
+  (inbox, supersededByState, alwaysShow) => {
+    return inbox.filter(i => (!i.isEmpty || alwaysShow.has(i.conversationIDKey)) &&
+        !supersededByState.get(i.conversationIDKey)).map(i => i.conversationIDKey)
   }
-}
-
-function _filterInboxes (inboxes: List<InboxState>, supersededByState: SupersededByState, alwaysShow: Set<ConversationIDKey>): List<InboxState> {
-  // $FlowIssue with records and accessing things inside them
-  return inboxes.filter(i => (!i.isEmpty || alwaysShow.has(i.conversationIDKey)) && !supersededByState.get(i.conversationIDKey))
-}
+)
+const getRows = createImmutableEqualSelector(
+  [filteredInbox, getPending],
+  (inbox, pending) => I.List(pending.keys()).concat(inbox)
```

Comment:
This makes this component only aware of the list of keys. The rows themselves will be connected. This disconnects a row updating from affecting other rows!